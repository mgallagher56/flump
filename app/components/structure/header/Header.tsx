import { type FC, type ReactElement, useEffect, useState } from 'react';

import { Flex } from '@chakra-ui/react';
import ColorModeSwitch from '~/components/ColorModeSwitch';
import FLPBox from '~/components/core/structure/FLPBox';
import NavMenu from '~/components/navigation/NavMenu';
import UserLogin from '~/components/navigation/UserLogin';

import HomeLogo from '../../navigation/HomeLogo';
import { loginStyles, menuStyles, navStyles } from './styles';

interface HeaderProps {
  showColorModeSwitch?: boolean;
}

const Header: FC<HeaderProps> = ({ showColorModeSwitch = true }): ReactElement => {
  const [isApp, setIsApp] = useState(false);
  useEffect(() => {
    setIsApp(!!window.location.pathname.includes('/app'));
  }, []);

  return (
    <FLPBox as="header">
      <Flex as="nav" className={navStyles}>
        <FLPBox className={menuStyles}>
          <HomeLogo />
          {isApp ? (
            <NavMenu
              routes={[
                { key: 'dashboard', route: '/app' },
                { key: 'accounts', route: '/app/accounts' }
              ]}
              showAppLink={false}
            />
          ) : (
            <NavMenu
              routes={[
                { key: 'home', route: '/' },
                { key: 'about', route: '/about' },
                { key: 'contact', route: '/contact' }
              ]}
            />
          )}
        </FLPBox>
        <FLPBox className={loginStyles}>
          <UserLogin />
          {showColorModeSwitch && <ColorModeSwitch />}
        </FLPBox>
      </Flex>
    </FLPBox>
  );
};

export default Header;
