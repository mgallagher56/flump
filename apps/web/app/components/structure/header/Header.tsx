import { type FC, type ReactElement } from 'react';

import { Flex } from '@chakra-ui/react';
import { useLoaderData } from 'react-router';
import ColorModeSwitch from '~/components/ColorModeSwitch';
import FLPBox from '~/components/core/structure/FLPBox';
import HomeLogo from '~/components/navigation/HomeLogo';
import NavMenu from '~/components/navigation/NavMenu';
import UserLogin from '~/components/navigation/UserLogin';
import type { loader } from '~/root';

import { loginStyles, menuStyles, navStyles } from './styles';

interface HeaderProps {
  showColorModeSwitch?: boolean;
}

const Header: FC<HeaderProps> = ({ showColorModeSwitch = true }): ReactElement => {
  const { user } = useLoaderData<typeof loader>();
  return (
    <FLPBox as="header">
      <Flex as="nav" className={navStyles}>
        <FLPBox className={menuStyles}>
          <HomeLogo />
          <NavMenu
            routes={[
              { key: 'home', route: '/' }
              // { key: 'about', route: '/about' },
              // { key: 'contact', route: '/contact' }
            ]}
          />
          {!!user && (
            <NavMenu
              routes={[
                { key: 'dashboard', route: '/app' },
                { key: 'accounts', route: '/app/accounts' }
              ]}
            />
          )}
        </FLPBox>
        <FLPBox className={loginStyles}>
          <UserLogin />
          {showColorModeSwitch && <ColorModeSwitch />}
        </FLPBox>
      </Flex>
    </FLPBox>
  );
};

export default Header;
