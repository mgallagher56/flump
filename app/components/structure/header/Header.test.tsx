import { render } from '@testing-library/react';
import { describe, expect, test, vi } from 'vitest';
import Header from '~/components/structure/header/Header';

vi.mock('~/components/ColorModeSwitch', () => ({
  default: () => <div>ColorModeSwitch</div>
}));

describe('<Header />', () => {
  test('renders as expected when not in app route', () => {
    const { baseElement } = render(<Header />);
    expect(baseElement).toMatchSnapshot();
  });

  test('renders without ColorModeSwitch', () => {
    const { baseElement } = render(<Header showColorModeSwitch={false} />);
    expect(baseElement).toMatchSnapshot();
  });

  test('renders as expected when in app route', () => {
    Object.defineProperty(window.location, 'pathname', {
      value: '/app'
    });
    const { baseElement } = render(<Header />);
    expect(baseElement).toMatchSnapshot();
  });
});
