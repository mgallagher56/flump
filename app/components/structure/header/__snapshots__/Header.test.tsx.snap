// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Header /> > renders as expected by default 1`] = `
<body>
  <div>
    <header
      class="css-0"
    >
      <nav
        class="d_flex border-bw_2px border-b_blue.500 border-bottom-style_solid items_center justify_space-between w_100% py_4 px_2 css-k008qs"
      >
        <div
          class="d_flex gap_4 items_center justify_start css-0"
        >
          <div>
            HomeLogo
          </div>
          <div>
            home
          </div>
        </div>
        <div
          class="d_flex gap_2 items_center justify_end css-0"
        >
          <div>
            UserLogin
          </div>
          <div>
            ColorModeSwitch
          </div>
        </div>
      </nav>
    </header>
  </div>
</body>
`;

exports[`<Header /> > renders correctly when user logged in 1`] = `
<body>
  <div>
    <header
      class="css-0"
    >
      <nav
        class="d_flex border-bw_2px border-b_blue.500 border-bottom-style_solid items_center justify_space-between w_100% py_4 px_2 css-k008qs"
      >
        <div
          class="d_flex gap_4 items_center justify_start css-0"
        >
          <div>
            HomeLogo
          </div>
          <div>
            home
          </div>
          <div>
            dashboard
          </div>
          <div>
            accounts
          </div>
        </div>
        <div
          class="d_flex gap_2 items_center justify_end css-0"
        >
          <div>
            UserLogin
          </div>
          <div>
            ColorModeSwitch
          </div>
        </div>
      </nav>
    </header>
  </div>
</body>
`;

exports[`<Header /> > renders without ColorModeSwitch 1`] = `
<body>
  <div>
    <header
      class="css-0"
    >
      <nav
        class="d_flex border-bw_2px border-b_blue.500 border-bottom-style_solid items_center justify_space-between w_100% py_4 px_2 css-k008qs"
      >
        <div
          class="d_flex gap_4 items_center justify_start css-0"
        >
          <div>
            HomeLogo
          </div>
          <div>
            home
          </div>
        </div>
        <div
          class="d_flex gap_2 items_center justify_end css-0"
        >
          <div>
            UserLogin
          </div>
        </div>
      </nav>
    </header>
  </div>
</body>
`;
